### DISCUSSION TITLE
Suzuka Oneshot Bootstrap Mode

### DISCUSSION THREAD

**Discussion by l-monninger** [movementlabsxyz/movement#469] [Ideas]
# Summary
As an alternative to the p2p with verification syncing discussed in #468 and the RFC linked therein, we can perhaps define a simpler version that precludes light running modes, but still allows trustless bootstrapping. This would be to download the entirety of state as a file, then populate the Merkle tree offline, verify its integrity, and check it matches the state root on-chain. This ensures the Aptos state is correct. 

Currently, the main challenge here would ensuring all state in Aptos DB is verifiable via the currently available JMT. This is also, arguably, the main challenge in #468 , so it's unclear that the implementation time gained would be significant.

## Threats
- **Big Bad State:** rather than verifying many small bits of data that might not be malicious, this approach requires downloading all data before verifying its correctness. Additionally, if this is done by files, not all data would be validated. Instead only that which is used in the JMT verification. While in an idea world, most of the unverified data could be considered inert, the attack surface is greatly increased.

**Comment by musitdev** [comment]
The state proof contains the JMT of all Tx since the beginning. Calculating the Merkle tree with the Tx in the db will take time, and I'm not sure, it's enough. The account ledger state is not part of the state proof. So even if the Merkle tree is correct, we can't be sure the account ledger is valid without re-excuting all the block from the beginning.
